src/cdk/scrolling/virtual-scroll-viewport.ts

```diff
@@ -321,6 +321,8 @@ export class CdkVirtualScrollViewport extends CdkScrollable implements OnInit, O
     const axis = isHorizontal ? 'X' : 'Y';
     const axisDirection = isHorizontal && isRtl ? -1 : 1;
     let transform = `translate${axis}(${Number(axisDirection * offset)}px)`;
+    // in appendOnly, we always start from the top
```

Comment:
I've moved this logic to the virtual-scroll-viewport and added. I'm wondering if we need to update the offset when  `to` is set to `to-end`. I've added an unit test for this case and the offset is correctly set. I think it's because we are recalculating the values with 'to-start' afterwards

Reply:
This looks good, thanks!